hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -488,64 +491,61 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
 	return filtered, nil
 }
 
-func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request) error {
+func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request) (*ctrl.Result, error) {
```

Comment:
Nit: pointer seems like extra annoying work to deref and handle nils, not sure why a value wouldn't suffice as with `Reconcile` itself

Reply:
The return values need to convey 3 cases

1. delete is complete without error (nil, nil)
2. delete did not error but is waiting and we need to requeue (result, nil)
3. delete encountered an error (nil|result, err)

If we don't make the return `result` a pointer, we can't distinguish between 1 and 2 right?

Reply:
It's not clear to me what the purpose of 1 is — unless you're coalescing multiple collected results from the call side (I don't think we should do this), then if `err` is `nil` I would expect the caller to use the non-pointer zero-value result.